lang/java/reef-runtime-azbatch/src/main/java/org/apache/reef/runime/azbatch/util/AzureStorageUtil.java

```diff
@@ -0,0 +1,129 @@
+      "DefaultEndpointsProtocol=https;AccountName=%s;AccountKey=%s";
+
+  private final CloudStorageAccount storageAccount;
+  private final CloudBlobClient blobClient;
+  private final CloudBlobContainer container;
+  private final String azureStorageContainerName;
+
+  @Inject
+  AzureStorageUtil(
+      @Parameter(AzureStorageAccountName.class) final String accountName,
+      @Parameter(AzureStorageAccountKey.class) final String accountKey,
+      @Parameter(AzureStorageContainerName.class) final String azureStorageContainerName)
+      throws URISyntaxException, InvalidKeyException, StorageException {
+
+    final String connectionString = getStorageConnectionString(accountName, accountKey);
+    this.storageAccount = CloudStorageAccount.parse(connectionString);
+    this.blobClient = this.storageAccount.createCloudBlobClient();
+    this.azureStorageContainerName = azureStorageContainerName;
+    this.container = this.blobClient.getContainerReference(this.azureStorageContainerName);
+    this.container.createIfNotExists();
+
+    LOG.log(Level.FINE, "Instantiated AzureStorageUtil connected to azure storage account: {0}", accountName);
+  }
+
+  /**
+   * Assemble a connection string from account name and key.
+   */
+  private static String getStorageConnectionString(final String accountName, final String accountKey) {
+    return String.format(AZURE_STORAGE_CONNECTION_STRING_FORMAT, accountName, accountKey);
+  }
+
+  public URI createFolder(final String folderName) throws IOException {
+    try {
+      final CloudBlockBlob jobFolderBlob = this.container.getBlockBlobReference(folderName);
+      return jobFolderBlob.getStorageUri().getPrimaryUri();
+    } catch (final StorageException | URISyntaxException e) {
+      throw new IOException("Unable to create job Folder", e);
+    }
+  }
+
+  public URI uploadFile(final URI folder, final File file) throws IOException {
+
+    LOG.log(Level.INFO, "Uploading [{0}] to [{1}]", new Object[]{file, folder});
+
+    try {
+      final CloudBlockBlob blob = this.container.getBlockBlobReference(String.format("%s/%s", folder, file.getName()));
+
+      try (final BufferedInputStream in = new BufferedInputStream(new FileInputStream(file))) {
+        blob.upload(in, file.length());
+      }
+
+      LOG.log(Level.FINE, "Uploaded to: {0}", blob.getStorageUri().getPrimaryUri());
+
+      final String sas = blob.generateSharedAccessSignature(getSharedAccessBlobPolicy(), null);
+      final String uri = blob.getStorageUri().getPrimaryUri().toString();
+      return new URI(uri + "?" + sas);
+
+    } catch (final URISyntaxException | StorageException | InvalidKeyException e) {
+      throw new IOException(e);
+    }
+  }
+
+  private SharedAccessBlobPolicy getSharedAccessBlobPolicy() {
+
+    Date now = Calendar.getInstance().getTime();
+
+    Calendar calendar = Calendar.getInstance();
+    calendar.add(Calendar.MINUTE, SAS_TOKEN_VALIDITY_MINUTES);
```

Comment:
nit: Odd for this to be static. Maybe a default parameter?

Reply:
it's a java convention to make hard-coded variables static. Not sure of why exactly that came up as standard practice, but I see that time and again in Java code.